pkg/appgw/requestroutingrules.go

```diff
@@ -38,7 +38,7 @@ func (c *appGwConfigBuilder) pathMaps(ingress *v1beta1.Ingress, rule *v1beta1.In
 	for pathIdx := range rule.HTTP.Paths {
 		path := &rule.HTTP.Paths[pathIdx]
 		backendID := generateBackendID(ingress, rule, path, &path.Backend)
-		backendPool := c.backendPoolMap[backendID]
+		backendPool := c.NewBackendPoolMap()[backendID]
```

Comment:
One tiny optimization we could do is - we could create `backendPools := c.NewBackendPoolMap()` before we enter in this `for pathIdx := range rule.HTTP.Paths {` loop; then here on this line - lookup `backendPool := backendPools [backendID]`

Another alternative would be to memoize - but that seems a bit more involved